src/Pim/index.tsx

```diff
@@ -395,6 +398,30 @@ class Pim extends React.PureComponent {
     const { collectionsAddressBook, collectionsCalendar, collectionsTaskList, addressBookItems, calendarItems, taskListItems } = itemsSelector(this.props);
     const [contacts, events, tasks] = itemValuesSelector(this.props);
 
+    let birthdays: EventType[] = [];
+
+    if (store.getState().settings.showBirthdayCalendar) {
+      birthdays = contacts.filter((c) => c.bday).map((c) => {
+        const event = new EventType();
+        const bdayTime = parseDate(c.comp.getFirstProperty('bday'));
+        if (bdayTime !== {} && bdayTime.month !== undefined) {
+          const year = bdayTime.year ?? 1900;
+          const month = bdayTime.month;
+          const day = bdayTime.day;
+          const date = new Date(year, month, day);
+
+          event.summary = `${c.fn}'s Birthday`;
+          event.startDate = timeNativeToVobject(date, true);
+          const rrule: ICAL.RecurData = {
+            freq: 'YEARLY',
+            interval: 1,
+          };
+          event.rrule = new ICAL.Recur(rrule);
+        }
+        return event;
+      }).filter((ev) => ev.uid !== undefined);
```

Comment:
Maybe I'm blind, but I can't see you setting the event uid at all.
Additionally, you probably want to change this filter to `filter(isDefined)` using this function: https://github.com/etesync/ios/blob/master/src/helpers.tsx#L110
and just return undefined as the first step once you realise this event can't be parsed. Without even creating the event in the first place.